clients/admin-ui/src/features/config-wizard/ScanResultsForm.tsx

```diff
@@ -33,26 +36,33 @@ const ALL_COLUMNS: ColumnMetadata[] = [
 const ScanResultsForm = () => {
   const systems = useAppSelector(selectSystemsForReview);
   const dispatch = useAppDispatch();
+  const router = useRouter();
   const {
     isOpen: isWarningOpen,
     onOpen: onWarningOpen,
     onClose: onWarningClose,
   } = useDisclosure();
+  const [upsertSystems] = useUpsertSystemsMutation();
   const [selectedSystems, setSelectedSystems] = useState<System[]>(systems);
+  const features = useFeatures();
   const [selectedColumns, setSelectedColumns] =
     useState<ColumnMetadata[]>(ALL_COLUMNS);
 
+  const createSystems = async () => {
+    await upsertSystems(selectedSystems);
+  };
+
   const confirmRegisterSelectedSystems = () => {
     dispatch(chooseSystemsForReview(selectedSystems.map((s) => s.fides_key)));
-    dispatch(changeStep());
+    createSystems();
+    return features.plus ? router.push(`/datamap`) : router.push(`/system`);
```

Comment:
I don't think you need a `return` statement here?

Reply:
I think I was getting a linting error since I'm using an arrow function here that it needs to return something at the end of the fn 🤔

Reply:
Your other comment made me curious, so I'll take another look